fullstack-test.sh

```
@@ -235,13 +242,27 @@ if [ "$CLUSTER" = "localnet" ]; then
+        [ -t 1 ] && stty sane < /dev/tty 2>/dev/null || true
         break
       fi
+      if ! kill -0 "$EPHEMERAL_VALIDATOR_PID" 2>/dev/null; then
+        echo -e "${RED}Error: ephemeral-validator exited before becoming ready${NC}"
+        if [ -f /tmp/ephemeral-validator.log ]; then
+          echo "Startup log (/tmp/ephemeral-validator.log):"
+          sed -n '1,80p' /tmp/ephemeral-validator.log
+        fi
+        exit 1
+      fi
       if [ $i -eq 60 ]; then
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)